inc/zoo/map/RobinHood.h

```diff
@@ -0,0 +1,412 @@
+
+        // I (Eddie) believe an early exit of not having "sames" is not
+        // profitable here
+        // filter the needle, haystack equals by being before the deadline
+        // (by the way, the deadline means that slot can't be equal)
+        auto haystackRichersAsNumber = haystackStrictlyRichers.value();
+        // because we make the assumption of LITTLE ENDIAN byte ordering,
+        // we're interested in the elements up to the first haystack-richer
+        auto deadline = swar::isolateLSB(haystackRichersAsNumber);
+        // to analize before the deadline, "maskify" it.  Remember, the
+        // deadline element itself can't be a potential match, it does
+        // not need preservation
+        auto deadlineMaskified = Metadata{deadline - 1};
+        auto beforeDeadlineMatches = sames & deadlineMaskified;
+        return {
+            deadline,
+            beforeDeadlineMatches
+        };
+    }
+
+    /*! \brief converts the given starting PSL and reduced hash code into a SWAR-ready needle
+    
+    The given needle would have a PSL as the starting (PSL + 1) in the first slot, the "+ 1" is because
+    the count starts at 1, in this way, a haystack PSL of 0 is always "richer"
+    */
+    constexpr static auto makeNeedle(U startingPSL, U hoistedHash) {
+        constexpr auto Ones = meta::BitmaskMaker<U, 1, Width>::value;
+        constexpr auto Progression = Ones * Ones;
+        auto core = startingPSL | (hoistedHash << PSL_Bits);
+        auto broadcasted = broadcast(Metadata(core));
+        auto startingPSLmadePotentialPSLs = Metadata(Progression) + broadcasted;
+        return startingPSLmadePotentialPSLs;
+    }
+
+    template<int Misalignment, typename KeyComparer>
+    constexpr static auto
+    unalignedFind(
+        U hoistedHash,
+        Metadata *base,
+        int homeIndex,
+        const KeyComparer &kc
+    ) noexcept {
+        constexpr auto Ones = meta::BitmaskMaker<U, 1, Width>::value;
+        constexpr auto Progression = Metadata{Ones * Ones};
+
+        MisalignedGenerator<Metadata, 8*Misalignment> p{base};
+        auto index = homeIndex;
+        auto needle = makeNeedle(0, hoistedHash);
+        for(;;) {
+            auto result = potentialMatches(needle, *p);
+            auto positives = result.potentialMatches;
+            auto deadline = result.deadline;
+            while(positives.value()) {
+                auto matchSubIndex = positives.lsbIndex();
+                auto matchIndex = index + matchSubIndex;
+                if(kc(matchIndex)) {
+                    return std::tuple(true, matchIndex);
+                }
+                positives = Metadata{clearLSB(positives.value())};
+            }
+            if(deadline) {
+                auto position = index + Metadata{deadline}.lsbIndex();
+                return std::tuple(false, position);
+            }
+            // Skarupke's tail allows us to not have to worry about the end
```

Comment:
I do think we need to handle tail and no tail at some point in the future (since no tail might be better for smaller tables). That's after presentation though, I suspect.

Reply:
And shouldn't the metdata provider have the concept of the skarupke tail, not RH_Backend? Would that require it keeping track of what index results are in (instead of RH_Backend)

Reply:
The metadata provider ought not to have knowledge of the tail: it ought to only have knowledge of the size of the metadata it can provide.
The type of tail is directly bound to the meaning of the flavor of the RH table, then I think the natural place is the backend